internal/backends/winit/renderer/skia.rs

```diff
@@ -168,13 +175,13 @@ impl i_slint_core::renderer::Renderer for SkiaRenderer {
 
         let window = WindowInner::from_pub(window_adapter.window());
 
-        let scale_factor = window.scale_factor();
+        let scale_factor = ScaleFactor::new(window.scale_factor());
 
-        let max_width = text_input.width() * scale_factor;
-        let max_height = text_input.height() * scale_factor;
-        let pos = pos * scale_factor;
+        let max_width = LogicalLength::new(text_input.width());
```

Comment:
It used to be physical before. Looks like a bug now .

The text layout uses physical coordinate. Maybe using the proper physical unit in the text layout would catch this bug

Reply:
Hah yeah, it looks like a bug, but fortunately it's correct (and the compiler told me as I was refactoring this). The function in `textlayout.rs` used to take a `max_width: Option<Coord>` but now takes a `max_width: Option<LogicalLength>` It also took and still takes a scale factor, because of the logical sizes in the FontRequest. My "final" series (after femtovg) changes `FontRequest` to property represent that, and then I'll change `text layout.rs` to operate only on physical.